app/components/submission/header.js

```diff
@@ -29,20 +29,43 @@ export default class SubmissionHeaderComponent extends Component {
   @tracked comment;
 
   @tracked selectedAgenda;
+  @tracked selectedMeeting;
 
   constructor() {
     super(...arguments);
     this.loadAgenda.perform();
   }
 
   loadAgenda = task(async () => {
-    const meeting = await this.args.submission.meeting;
-    if (meeting?.id) {
-      this.selectedAgenda = await this.store.queryOne('agenda', {
-        'filter[created-for][:id:]': meeting.id,
-        'filter[:has-no:next-version]': true,
-      });
-    }
+    if (this.args.submission) {
+      const meeting = await this.args.submission.meeting;
+      if (meeting?.id) {
+        this.selectedMeeting = meeting;
+        this.selectedAgenda = await this.store.queryOne('agenda', {
+          'filter[created-for][:id:]': meeting.id,
+          'filter[:has-no:next-version]': true,
+        });
+      } else {
```

Comment:
realized this after PR review.
SelectedAgenda will usually be wrong for cabinet worker?
They should submit for the latest agenda, but the latest agenda (designagenda) is not propagated yet.
So we should get from ember store for kanselarij editors and from custom endpoint for cabinet workers

Reply:
hmm, this is a bit confusing.
We use this header component in 2 places.

1. new-submission from case view (either new submission or BIS submission)
    here we never have a submission yet, so `selectedAgenda is null`, even for BIS update we don't check what the current meeting of the previous submission was, that happens on creation in a different component (mails are sent there aswell)
    -we only pass it as `defaultAgenda` when it is a new submission (not BIS). In that case, we never need a default because the user has yet to choose an agenda.
    -in case of a BIS, we open a different modal and don't even use the `selectedAgenda`
2. when editing a submission
   - editing an existing submission does need the meeting, because we want to fill in the agenda (latest version) for secretarie to match the choice of meeting that happened on initial creation of the submission.
   
   So my initial comment is correct, we load the incorrect agenda for cabinet worker, but the agenda itself is never used so it doesn't really matter.

Reply:
tested it to confirm.
in case 2 (editing existing) 
Secretarie profile sees agenda B as `selectedAgenda` (for use in Proposableagendamodal)
Cabinet profile sees agenda A as `selectedAgenda` (but not used)

Reply:
fixed